src/db/migrations/20230907143204_add_onerep_scanresults_table.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export async function up(knex) {
+  await knex.schema.alterTable("onerep_scans", table => {
+    table.unique("onerep_scan_id");
+  });
+
+  await knex.schema
+    .createTable("onerep_scan_results", table => {
+      table.increments("id").primary();
+      table.integer("onerep_scan_result_id").notNullable();
+      table.integer("onerep_scan_id").references("onerep_scans.onerep_scan_id").notNullable();
+      table.string("link").notNullable();
+      table.integer("age").nullable();
+      table.string("data_broker").notNullable();
+      table.integer("data_broker_id").notNullable();
+      table.jsonb("emails").notNullable();
+      table.jsonb("phones").notNullable();
+      table.jsonb("addresses").notNullable();
+      table.jsonb("relatives").notNullable();
+      table.string("first_name").notNullable();
+      table.string("middle_name").nullable();
+      table.string("last_name").notNullable();
+      table.string("status").notNullable();
+      table.timestamp("created_at").defaultTo(knex.fn.now());
+      table.timestamp("updated_at").defaultTo(knex.fn.now());
+      table.index("onerep_scan_id");
+      table.index("onerep_scan_result_id");
+    });
+
+  const scanRows = await knex("onerep_scans")
+    .select("id", "onerep_scan_id", "onerep_scan_results");
+
+  const scanResultRows = scanRows.map(scan => {
+    return scan.onerep_scan_results.data.map(scanResult => {
+      const rowToInsert = {
+        onerep_scan_result_id: scanResult.id,
+        onerep_scan_id: scan.onerep_scan_id,
+        link: scanResult.link,
+        age: typeof scanResult.age === "string" ? Number.parseInt(scanResult.age, 10) : null,
```

Comment:
So from the test data, we get `null` for all ages, but I wonder if this is going to be problematic when we get some real data that has `age: number`.. and this would nullify that

Reply:
Yeah, that's fair. For what it's worth, [the OneRep API docs](https://docs.onerep.com/#operation/getScanResults) do say they return a string. I can add another check for `number` if you prefer, though I'm also a bit wary of handling cases that shouldn't happen (and thus can't be tested).

Reply:
We can go with the API then, good call